osidb/serializer.py

```diff
@@ -2394,53 +2398,21 @@ def get_affects(self, obj):
         # Materialize affect list so we don't re-run the query
         affects = list(affects_v1)
 
-        # Build list of already queried entities to cascade to the next serializer
-        tracker_uuid_set = set()
-        affects_by_tracker = defaultdict(list)
-        for a in affects:
-            ids = getattr(a, "all_tracker_ids", None) or []
-            for t in ids:
-                tracker_uuid_set.add(t)
-                affects_by_tracker[t].append(a.uuid)
-
-        # Build a single alerts map for flaw, affects, and trackers
-        try:
-            object_ids = set()
-            last_validated_by_object = {}
-
-            # Flaw
-            object_ids.add(str(obj.uuid))
-            last_validated_by_object[str(obj.uuid)] = getattr(
-                obj, "last_validated_dt", timezone.now()
-            )
-
-            # Affects (using materialized list)
-            for a in affects:
-                object_ids.add(str(a.uuid))
-                last_validated_by_object[str(a.uuid)] = (
-                    getattr(a, "last_validated_dt", None) or timezone.now()
-                )
-
-            # Trackers from affects
-            if tracker_uuid_set:
-                tracker_rows = Tracker.objects.filter(
-                    uuid__in=list(tracker_uuid_set)
-                ).values("uuid", "last_validated_dt")
-                for row in tracker_rows:
-                    object_ids.add(str(row["uuid"]))
-                    last_validated_by_object[str(row["uuid"])] = (
-                        row["last_validated_dt"] or timezone.now()
-                    )
-
+        # Bulk prefetch alerts for all affects to avoid N+1 queries
+        if affects:
+            affect_uuid_set = {str(a.uuid) for a in affects}
+            # Alerts are linked to Affect model (not AffectV1), so use Affect ContentType
+            affect_content_type = ContentType.objects.get_for_model(Affect)
```

Comment:
you don't need to filter by content type, we use uuids so simply looking up by object_id should suffice